[Oscar Su Zhen Hong] Duke Increments

oscarsu97 opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

j-lum committed:
Change file mode on `gradle` to be executable

j-lum committed:
Merge pull request #12 from j-lum/gradle+x

Change file mode on `gradle` to be executable (#9)

j-lum committed:
Add configuration for console applications

Gradle defaults to an empty stdin which results in runtime exceptions
when attempting to read from `System.in`. Let's add some sensible
defaults for students who may still need to work with the standard
input stream.

j-lum committed:
Merge pull request #13 from j-lum/javaexec

Add configuration for console applications

oscarsu97 committed:
Level-1

oscarsu97 committed:
Level 2.Add,List

oscarsu97 committed:
Level3. Mark as Done

oscarsu97 committed:
Level3.Mark as Done

oscarsu97 committed:
Level 4.ToDos, Events, Deadlines

oscarsu97 committed:
Level 5.Handle Errors

oscarsu97 committed:
Level 5.Handle Errors

oscarsu97 committed:
Level 5.Handle Errors

oscarsu97 committed:
Level 5.Handle Errors

oscarsu97 committed:
Level 6. Delete

oscarsu97 committed:
Level 7. Able to load data from duke.txt and update the text file after user is done.

oscarsu97 committed:
commit data/duke.txt

oscarsu97 committed:
clean up the code of Duke, Deadline, Event, Task class

oscarsu97 committed:
Level 8. store dateTime string variable as LocalDateTime in Event and Deadline class

oscarsu97 committed:
Merge conflict resolved between master and level 7 branch

oscarsu97 committed:
remove error in printing output

oscarsu97 committed:
Add Parser, Storage, Ui Class for refactoring

oscarsu97 committed:
Update duke.txt with proper format

oscarsu97 committed:
Divide classes into packages

oscarsu97 committed:
Add TaskListTest and EventTest class for testing

oscarsu97 committed:
Update library file for testing

oscarsu97 committed:
Add JavaDoc for all classes

oscarsu97 committed:
Refactor code to follow coding standard

oscarsu97 committed:
Find a task by searching for a keyword

'find' command added in Parser class

oscarsu97 committed:
Merge branch-A-JavaDoc and master branch

Add JavaDoc for all classes

oscarsu97 committed:
Merge branch 'branch-A-CodingStandard'

Tweak the code to comply with a chosen coding standard

oscarsu97 committed:
Merge branch 'branch-Level-9'

Parser also check if 'find' is in command.
TaskList class is able to find task that matches keyword.

oscarsu97 committed:
Merge branch 'gradle'

oscarsu97 committed:
Add gradle support to project

Able to checkstyle, make executable jar file
Create a gradle wrapper.

tishyakhanna97 reviewed (commented):
Coding standards - Could change package names to lowercase to adhere to style guide (lowercase to avoid conflict with the names of classes/interfaces)

tishyakhanna97 reviewed (commented):
Good use of IndexOffBoundException, very useful for this bot

oscarsu97 committed:
Add a GUI to to Duke

oscarsu97 committed:
Merge branch 'branch-A-JavaFx'

Abhiman2211 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,253 @@
+     *
+     * @return a list of the updated tasks
+     */
+    public String parse(String input) {
```
Perhaps for this method, you could break down the code within each if statements and make methods to handle each chunk of code? This would improve the readability of the code and make it more OOP overall, in my opinion.

Abhiman2211 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,53 @@
+     * @param description description of task
+     * @param dateTime    date and time to be completed
+     */
+    public Deadline(String description, String dateTime) {
```
I like how well you have handled the creation of the deadline. Maybe you could parse the date and time outside the constructor to make the method cleaner?

Abhiman2211 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,105 @@
+     * @param keyword the string to look for
+     * @return ArrayList that contain all tasks that match the keyword
+     */
+    public ArrayList<Task> findMatchingTasks(String keyword) {
```
I like your findMatchingTasks method! It covers all possible scenarios very nicely. Well done!

oscarsu97 committed:
Fix error in Level 10

Duke class able to run as per usual instead of
just Launcher class

oscarsu97 committed:
Add assert feature

oscarsu97 committed:
Merge branch 'master' from remote repository

oscarsu97 committed:
Fix bug in assert feature

oscarsu97 committed:
Refactor code to improve code quality (#5)

oscarsu97 committed:
Merge branch 'master' with local master branch

Refactor code for better code quality

oscarsu97 committed:
Add Statistic class

Statistic class that calculate the number
of deadline, event, todo tasks and the
percentage of work done

oscarsu97 committed:
Restructure code for better code quality

Parser class returns specific command class
instead of string.

oscarsu97 committed:
Add travis file

oscarsu97 committed:
Fix error in EventTest class

oscarsu97 committed:
Add cache feature to travis yml file

Speed up build process

oscarsu97 committed:
Fix minor error in exception handling

oscarsu97 committed:
Edit user interface

oscarsu97 committed:
Set theme jekyll-theme-slate

oscarsu97 committed:
Update README.md.

oscarsu97 committed:
Merge branch 'master' of https://github.com/oscarsu97/duke

oscarsu97 committed:
Improve code quality
